### PR TITLE
feat(role): add homeassistant

### PR DESCRIPTION
**Pull request by RaneyDazed**
# Description

This PR is for Home assistant, it is open source home automation that puts local control and privacy first. Its powered by a worldwide community of tinkerers and DIY enthusiasts. Perfect to run on a Raspberry Pi or a local server. Docker compose: 

```yaml
version: '3'
services:
  homeassistant:
    container_name: homeassistant
    image: "lscr.io/linuxserver/homeassistant:stable"
    volumes:
      - /opt/homeassistant:/config
      - /etc/localtime:/etc/localtime:ro
    restart: unless-stopped
    privileged: true
    network_mode: host
```

It connects to LAN devices just fine, and is connectable after initial setup on lan IP, ie `192.168.1.xx:8123`

# Doccumentations 

I will create the docs post haste!

# How Has This Been Tested?

I tested it many times on my local server, removing files and the directory multiple times. Salty gave me a hand with it a few times, figuring out what would allow host mode. Requires newest update of sb and traefik. To allow for the role to work with docker network mode: host.

### CODE REVIEW

**Comment by owine** [comment]
Tag not added to sandbox.yml

**Comment by RaneyDazed** [comment]
Yup I'd realized that a second ago, fixing rn

**Comment by RaneyDazed** [comment]
Sorry I'd reverted back to master and didn't realize it. Should be set now.

**Comment by keldian** [comment]
The right name is _home-assistant_, not _homeassistant_.

File: roles/homeassistant/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+################################
+
+# Container
+homeassistant_docker_container: "{{ homeassistant_name }}"
+
+# Image
+homeassistant_docker_image_pull: true
+homeassistant_docker_image_tag: "latest"
+homeassistant_docker_image: "ghcr.io/home-assistant/home-assistant:{{ homeassistant_docker_image_tag }}"
+
```
Comment (JigSawFr) [inline roles/homeassistant/defaults/main.yml:66]: Please add healthcheck block here :)

**Review by JigSawFr — CHANGES_REQUESTED** [review]

**Comment by owine** [comment]
If someone runs this role on a remote server in a datacenter, rather on a local LAN, do we introduce any issues?

**Review by owine — CHANGES_REQUESTED** [review]
If someone runs this role on a remote server in a datacenter, rather on a local LAN, do we introduce any issues?

**Comment by RaneyDazed** [comment]
> If someone runs this role on a remote server in a datacenter, rather on a local LAN, do we introduce any issues?

I will run it on my hetzner and see. I don't forsee any issues, but it wont do you any good really. What issues did you have in mind?

**Comment by JigSawFr** [comment]
@owine suggested to have a global var like home-use: true/false in settings of saltbox and block some roles depending of the value to avoid an issue, same for homebridge, pi-hole, etc
Another way is to not publish ports, and assume that user will use a vpn (like wireguard) container to consume them. (Or manually override to publish port).

**Comment by RaneyDazed** [comment]
Looks like I need to figure out this rebase thing before this will go anywhere

**Comment by RaneyDazed** [comment]
So where are we with homeassistant @JigSawFr @owine 

Is there something I'm supposed to do to somehow set it local only? I haven't installed on my remote server yet. I can do that then what am I looking for? anything in particular?

**Comment by RaneyDazed** [comment]
We can always put it on hold if necessary if we are waiting for some kind of true false local thing?

**Comment by owine** [comment]
I don’t care about a toggle, just want to make sure this won’t cause issues on a remote servers.

**Comment by RaneyDazed** [comment]
Tested on my remote server and didn't have any trouble. Loaded up just fine, logged in made a user etc.

**Pull request merged**